xsc/services/analyticsevent.go

```diff
@@ -0,0 +1,121 @@
+package services
+
+import (
+	"encoding/json"
+	"github.com/jfrog/jfrog-client-go/auth"
+	"github.com/jfrog/jfrog-client-go/http/jfroghttpclient"
+	"github.com/jfrog/jfrog-client-go/utils"
+	"github.com/jfrog/jfrog-client-go/utils/errorutils"
+	"github.com/jfrog/jfrog-client-go/xray/services"
+	"net/http"
+)
+
+type AnalyticsEventService struct {
+	client     *jfroghttpclient.JfrogHttpClient
+	XscDetails auth.ServiceDetails
+}
+
+func NewAnalyticsEventService(client *jfroghttpclient.JfrogHttpClient) *AnalyticsEventService {
+	return &AnalyticsEventService{client: client}
+}
+
+// GetXscDetails returns the Xsc details
+func (vs *AnalyticsEventService) GetXscDetails() auth.ServiceDetails {
+	return vs.XscDetails
+}
+
+// AddGeneralEvent add general event in Xsc and returns msi generated by Xsc.
+func (vs *AnalyticsEventService) AddGeneralEvent(event XscAnalyticsGeneralEvent) (string, error) {
+	httpDetails := vs.XscDetails.CreateHttpClientDetails()
+	requestContent, err := json.Marshal(event)
+	if err != nil {
+		return "", errorutils.CheckError(err)
+	}
+	resp, body, err := vs.client.SendPost(vs.XscDetails.GetUrl()+"api/v1/event", requestContent, &httpDetails)
+	if err != nil {
+		return "", err
+	}
+	if err = errorutils.CheckResponseStatus(resp, http.StatusCreated); err != nil {
+		return "", errorutils.CheckError(errorutils.GenerateResponseError(resp.Status, utils.IndentJson(body)))
+	}
+	var response XscAnalyticsGeneralEventResponse
+	err = json.Unmarshal(body, &response)
+	return response.MultiScanId, errorutils.CheckError(err)
+}
+
+// UpdateGeneralEvent update finalized analytics metrics info of an existing event.
+func (vs *AnalyticsEventService) UpdateGeneralEvent(event XscAnalyticsGeneralEventFinalize) error {
+	httpDetails := vs.XscDetails.CreateHttpClientDetails()
+	requestContent, err := json.Marshal(event)
+	if err != nil {
+		return errorutils.CheckError(err)
+	}
+	resp, body, err := vs.client.SendPut(vs.XscDetails.GetUrl()+"api/v1/event", requestContent, &httpDetails)
```

Comment:
```suggestion
	resp, body, err := vs.client.SendPut(vs.XscDetails.GetUrl()+xscEventAPI, requestContent, &httpDetails)
```
Use const for API EP